packages/query-core/src/queryClient.ts

```diff
@@ -366,92 +366,52 @@ export class QueryClient {
 
   setQueryDefaults(
     queryKey: QueryKey,
-    options: Omit<QueryObserverOptions<unknown, any, any, any>, 'queryKey'>,
+    options: Partial<
+      Omit<QueryObserverOptions<unknown, any, any, any>, 'queryKey'>
+    >,
   ): void {
-    const result = this.#queryDefaults.find(
-      (x) => hashKey(queryKey) === hashKey(x.queryKey),
-    )
-    if (result) {
-      result.defaultOptions = options
-    } else {
-      this.#queryDefaults.push({ queryKey, defaultOptions: options })
-    }
+    this.#queryDefaults.set(hashKey(queryKey), {
+      queryKey,
+      defaultOptions: options,
+    })
   }
 
   getQueryDefaults(
-    queryKey?: QueryKey,
+    queryKey: QueryKey,
   ): QueryObserverOptions<any, any, any, any, any> | undefined {
```

Comment:
the new code now garanties you'll at least return an empty object. so maybe just return :  
```suggestion
  ): QueryObserverOptions<any, any, any, any, any> {
```